### ISSUE TITLE
`snowflake_grant_privileges_to_role` removing OWNERSHIP from `privileges` array

### ISSUE THREAD

**Issue by davware** [Snowflake-Labs/terraform-provider-snowflake#2190]
**Provider Version**
0.73.0
**Terraform Version**
1.1.9
**Describe the bug**
When I create a `snowflake_grant_privileges_to_role` with `privileges` set to `["OWNERSHIP"]`, it is automatically removed and set to `[]`, resulting in every subsequent plan/apply trying to add the `"OWNERSHIP"` privilege back. 

For context, I am looping through a yaml file to create my resources in my `.tf` file like so:
```
resource snowflake_grant_privileges_to_role new_warehouse_grants {
    for_each = local.warehouse_grants  *yaml file used for looping*
    on_account_object {
        object_type = "WAREHOUSE"
        object_name = each.value.warehouse_name
    }
    role_name = each.value.role_name
    privileges = each.value.privileges
    with_grant_option = false
}
```

where the local.warhouse_grants file looks something like this: 
```
CPS_WH|OWNERSHIP:
  privileges:
  - OWNERSHIP
  role_name: SYSADMIN
  warehouse_name: CPS_WH
CPS_WH|PRODUCER_CPS_ROLE:
  privileges:
  - MONITOR
  - USAGE
  role_name: PRODUCER_CPS_ROLE
  warehouse_name: CPS_WH
```
Note, for all privileges other than `OWNERSHIP` I define all privileges for a given role in the same resource block, however when I try to do that with `OWNERSHIP` Snowflake throws an error saying `Cannot specify OWNERSHIP in privilege list.`, thus I had to define `OWNERSHIP` on its own. This doesn't seem to be happening for any other permissions, only `OWNERSHIP`. Any help is much appreciated, thanks!

**Expected behavior**

There should be no changes to the resource, since no changes were made to the TF configs

**Code samples and commands**

Example `terraform plan` snippet attempting to add OWNERSHIP back to `privileges`:
```
  # module.prism_grants.snowflake_grant_privileges_to_role.new_warehouse_grants[SYSTEM_YUKON_TABLEAU_WH|OWNERSHIP] will be updated in-place
! resource snowflake_grant_privileges_to_role new_warehouse_grants {
        id                = SYSADMIN|OWNERSHIP|false|false|false|true|false|false|false|false|WAREHOUSE|SYSTEM_YUKON_TABLEAU_WH||false||false|
      ! privileges        = [
          + OWNERSHIP,
        ]
        # (4 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }

  # module.prism_grants.snowflake_grant_privileges_to_role.new_warehouse_grants[UPNEST_WH|OWNERSHIP] will be updated in-place
! resource snowflake_grant_privileges_to_role new_warehouse_grants {
        id                = SYSADMIN|OWNERSHIP|false|false|false|true|false|false|false|false|WAREHOUSE|UPNEST_WH||false||false|
      ! privileges        = [
          + OWNERSHIP,
        ]
        # (4 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }

Plan: 0 to add, 101 to change, 0 to destroy.
```

output of running `terraform -chdir=terraform/dev_snowflake state show 'module.dev_grants.snowflake_grant_privileges_to_role.new_warehouse_grants["CPS_WH|OWNERSHIP"]'` (shows OWNERSHIP being removed from `privileges` array):
```
# module.dev_grants.snowflake_grant_privileges_to_role.new_warehouse_grants["CPS_WH|OWNERSHIP"]:
resource "snowflake_grant_privileges_to_role" "new_warehouse_grants" {
    all_privileges    = false
    id                = "SYSADMIN|OWNERSHIP|false|false|false|true|false|false|false|false|WAREHOUSE|CPS_WH||false||false|"
    on_account        = false
    privileges        = []
    role_name         = "SYSADMIN"
    with_grant_option = false

    on_account_object {
        object_name = "CPS_WH"
        object_type = "WAREHOUSE"
    }
}
```

**Comment by davware** [comment]
As I continue to move different grants to the new `snowflake_grant_privileges_to_role` resource, I've noticed that `OWNERSHIP` is not revoked for _all_ objects. So far I've only noticed this issue with warehouses and schemas. For the time being, I've split out the `OWNERSHIP` grants from the rest of them, and added 
```
ignore_changes = [
            privileges
        ]
```
as a workaround.

**Comment by sfc-gh-jcieslak** [comment]
Hey @davware @sfc-gh-ckite
Right now, we're currently working on designing / re-designing grants, and from our analysis, we decided to drop support for granting ownership for a few reasons which will be documented and soon published to the repository to give everyone context on our decision, but to give you a quick sum up: we can grant ownership, but ownership is not a privilege, it's a complete ownership transition which cannot be queried and in the Terrafrom plugin after we granted the ownership you're asking then we have to check the current state on your Snowflake and that cannot be performed, because again there's no clear way of querying those kinds of information. Plus there can be some cases where you grant ownership to the object that is outside of the Terraform reach and when you delete the defined grant resource it (the ownership) can't be revoked because the Terraform would have to have access to this object.

**tldr** Please if possible find another way of granting access to a given resource with **snowflake_grant_privileges_to_role**, because granting ownership will be deprecated and probably eventually removed.

**Comment by davware** [comment]
Hey @sfc-gh-jcieslak, thanks for your detailed explanation. I'm wondering if there are any plans to create more "ownership grant resources" similar to `snowflake_role_ownership_grant` and `snowflake_user_ownership_grant`? Or is the plan that ownership will need to be managed outside of terraform? Thanks!

**Comment by sfc-gh-jcieslak** [comment]
@davware For now, we don't (and probably won't) have any plans on adding new resources that would grant ownership and the decision will be backed by the mentioned document we should publish soon. We checked that in some cases it's not feasible to manage such resources within the Terraform plugin. Right now, we are focusing on supporting all of the grant features provided by Snowflake (and possible to manage within the Terraform) and we're designing new missing resources/data sources and re-designing **snowflake_grant_privileges_to_role** to fix some of its issues/pain points. After the design proposal is finished and everyone on the team has their word, we should know which grants are feasible and which are not to manage within the Terraform, but for now, I can already tell you that it probably won't be added.

**Comment by davware** [comment]
@sfc-gh-jcieslak understood, I'll look out for that document. Thanks for all the info!

**Comment by jelledv** [comment]
@sfc-gh-jcieslak I am wondering why can't you query ownership? I am looking forward to reading the document. I am hoping all grants are still manageable through Terraform :)

**Comment by sfc-gh-jcieslak** [comment]
@jelledv @davware
Hey, I would like to emphasize that we're in a phase of seeing what are the available options and what we can do. I said that we probably won't have a grant privilege resource and the keyword would be "probably". Right now, I would say my view on the topic changed, and probably that was my mistake in communicating (I apologize that I might've caused confusion in the future of our provider). All I would like to say right now is to stay patient and stay tuned for more updates on the topic.

**Comment by sfc-gh-jcieslak** [comment]
Hey @davware @jelledv 
Here's the document I was talking about https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/docs/technical-documentation/grants_redesign_design_decisions.md. Additionally, the latest version includes grant ownership resource.
